bb_server/src/plugin/types/world/mod.rs

```diff
@@ -87,6 +88,10 @@ impl PWorld {
     self.inner.summon_meta(entity::Type::Item, pos.inner, meta);
   }
 
+  pub fn spawn_particle(&self, particle: &PParticle) {
+    self.inner.spawn_particle(particle.inner.clone());
```

Comment:
I think you were wondering about this, and it's fine. Panda does everything with references, so you're forced to `clone` in cases like this, which is cheap so it doesn't matter much.